src/Form.php

```diff
@@ -371,4 +443,26 @@ private function generateAttributeLabel(string $name): string
 
         return $this->inflector->camel2words($name, true);
     }
+
+    private function propertyReader(string $attribute)
+    {
+        $getter = fn ($class, $attribute) => $class->$attribute ?? null;
```

Comment:
`$value = $this->$attribute`?

Reply:
Closure#bind() basically allows you to get an instance of a closure basically like adding APIs to existing objects!

Reply:
Yeah, but why we need it?

Reply:
You should:
1. Get value from `$this->get{attribute}()` method if it exist
2. Or get value from `$this->{attribute}` if it exist
3. Or throw error

Reply:
Directly getValue() of the properties for getAttributes($attribute), it allows to use any name for the methods, for example if you have getError(), isError(), resolveError(), how are you going to get the method name for the property, that's why only for getAttributeValue () makes sense to directly access the property.

https://github.com/terabytesoftw/yii-form/blob/a1b5ac32ebbc48952d35c4b62a4e409f89c5cac7/src/Form.php#L455

Reply:
```php
class MyForm
{
  private string $time;
  public function getTime(): DateTimeInterface { ... }

  public function rules():
  {
    return ['time' => (new DateTimeValidator())->notToday()];
  }
}
```
What kind of value will we  have in the `DateTimeValidator` validator or others?

Reply:
the validator takes the values of getAttributesValue from DataSetInterface review tests.

Reply:
Another thing to think about that form will interact with widgets like `ActiveForm::class`, `ActiveField::class`, `AR`, `ORM` and custom widgets, where each developer has different ways of working, so working with the properties directly will be the best we will avoid many bugs , or more complex code to guess the names.